### DISCUSSION TITLE
RFC: Migrate content to be rendered only on the server/at build time

### DISCUSSION THREAD

**Discussion by brandonroberts** [analogjs/analog#1150] [Ideas]
Currently, the `@analog/content` package is what we use to render markdown routes and content pages. This takes advantage of libraries like Prism, marked, and support for Shiki recently added by @nartc.

The advantage of this approach is consistently between the two environments and uses Angular's upstream APIs.

The disadvantage of this approach is we're rendering the content twice, during prerendering, and again during client hydration. This is fine for packages like prismjs which works in both environments, but is constrained with tools like shiki where all the highlighting is expected to be done server-side only.

This proposal moves the rendering pipeline to be done only during server rendering and build time. During client hydration, we would only be consuming rendered content.

Advantages
 - Content is already rendered on the server
 - We don't have to ship the associated libraries to the browser (Prism, marked, etc)

**Comment by ajitzero** [comment]
How would this affect `ssr: false`/SSG cases?

**Use-case:** SSG for landing page, terms, privacy, blog, etc, and SPA for everything else. No explicit SSR/hydration cases.

**Reply by brandonroberts** [reply]
It doesn't affect that use case. `ssr: false` disables SSR for prerendering and runtime server rendering.
